src/values/ObjectValue.js

```diff
@@ -111,33 +111,43 @@ export default class ObjectValue extends ConcreteValue {
     "$WeakMapData",
     "$WeakSetData",
   ];
+  static trackedPropertyBindingNames = new Map();
 
   getTrackedPropertyNames(): Array<string> {
     return ObjectValue.trackedPropertyNames;
   }
 
   setupBindings(propertyNames: Array<string>) {
     for (let propName of propertyNames) {
-      let desc = { writeable: true, value: undefined };
-      (this: any)[propName + "_binding"] = {
-        descriptor: desc,
-        object: this,
-        key: propName,
-      };
+      let propBindingName = ObjectValue.trackedPropertyBindingNames.get(propName);
+      invariant(propBindingName !== undefined);
+      (this: any)[propBindingName] = undefined;
```

Comment:
I don't understand the need for this.

Reply:
To avoid string concatenation over and over.

Reply:
That part I understand. Why you need to create the property, I don't understand.

Reply:
It is probably better to have separate hidden classes for the non normal case. The fewer properties we require for the 99% case, the better.

Reply:
In any case, the properties were created before my change, and are still created after my change. If tinkering with the hidden classes here would add more performance benefits, then we can investigate that separately. But I don't want to block this PR on that question.